mos-platform/neo6502/api/neo/console.h

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 LLVM-MOS Project
+// Licensed under the Apache License, Version 2.0 with LLVM Exceptions.
+// See https://github.com/llvm-mos/llvm-mos-sdk/blob/main/LICENSE for license
+// information.
+
+#include <stdint.h>
+
+#ifndef _NEO_CONSOLE_H
+#define _NEO_CONSOLE_H
+
+#include <neo6502.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Read and remove a key press.
+ *
+ * @return ASCII key value; 0 if no key presses queued.
+ */
+__attribute__((leaf))
```

Comment:
This attribute doesn't do anything on functions where definitions are provided in the LTO unilt.